src/components/Modal/SearchModal.js

```diff
@@ -0,0 +1,107 @@
+import React, { useRef, useState, useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+import styled from 'styled-components';
+import SearchListModal from './SearchListModal';
+import TagModal from './TagModal';
+
+function SearchModal(props) {
+  const [isTag, setIsTag] = useState(true);
+  const [text, setText] = useState('');
+  const history = useHistory();
+  const inputRef = useRef(null);
+
+  // useEffect(() => {
+
+  // }, [props.isModal]);
+
+  const closeModal = e => {
+    props.clickSearch();
+  };
+
+  const clickInput = e => {
+    e.stopPropagation();
+  };
+
+  const handleChange = e => {
+    setText(e.target.value);
+  };
+
+  const handleKeyEvent = e => {
+    if (e.key === 'Enter') {
+      history.push(`/search?query=${text}`);
+      props.clickSearch();
+      e.preventDefault();
+    }
+  };
+
+  return (
+    <Dimmer onClick={closeModal}>
+      <SearchContainer>
+        <Wrapper>
+          <form>
+            <div>
+              <input
+                type="search"
+                placeholder="#태그,포지션 검색"
+                onChange={handleChange}
+                onClick={clickInput}
+                onKeyPress={handleKeyEvent}
+                ref={inputRef}
+              />
+              <i className="fas fa-search"></i>
+            </div>
+          </form>
+          {text !== '' ? <SearchListModal /> : <TagModal />}
```

Comment:
빈문자열은 falsy값입니다. 특성을 잘 이해하면 비교연산자 굳이 안써줘도 됩니다.
```suggestion
          {text ? <SearchListModal /> : <TagModal />}
```